### PR TITLE
Descriptive param names, typing, update docs, add pre-commit hooks

### PR DESCRIPTION
**Pull request by pzelasko**
Merging this before the CI/wheel build fix and will release together as 0.8.1

### CODE REVIEW

File: kaldialign/__init__.py [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
 
-    for sym in b:
-        bi.append(sym2int[sym])
+    for sym in hyp:
+        hypi.append(sym2int[sym])
 
-    return _kaldialign.edit_distance(ai, bi, sclite_mode)
+    ans = _kaldialign.edit_distance(refi, hypi, sclite_mode)
+    ans["ref_len"] = len(refi)
+    ans["err_rate"] = ans["total"] / len(refi)
```
Comment (boeddeker) [inline kaldialign/__init__.py:41]: FYI: This line introduces the bug, that edit_distance doesn't work for an empty reference, it raises a ZeroDivisionError.

Comment (pzelasko) [reply kaldialign/__init__.py:41]: right, I’ll fix that. We should probably return one of inf, nan, or None: WDYT?

Comment (boeddeker) [reply kaldialign/__init__.py:41]: Thanks.

Hmm. How about mirroring the behavior of numpy?
```python
>>> np.array([1]) / np.array([0])
array([inf])
```

Comment (pzelasko) [reply kaldialign/__init__.py:41]: Fixed in #21, releasing as v0.9.1

**Pull request merged**